src/Estimators/src/BaseEstimatorFromFootIMU.cpp

```diff
@@ -0,0 +1,263 @@
+{
+    m_input = input;                    //TO DO: check if this line goes wrong
+    return true;
+}
+
+bool BaseEstimatorFromFootIMU::advance()
+{
+    constexpr auto logPrefix = "[BaseEstimatorFromFootIMU::advance]";
+
+    m_isOutputValid = false;
+
+    if (!m_isInitialized)
+    {
+        log()->error("{} The estimator is not initialized.", logPrefix);
+        return false;
+    }
+
+    // Set the 4 foot vertices in World reference frame [dimensions in meters]
+    m_p1 << +m_footWidth / 2, +m_footLength / 2, 0;
+    m_p2 << -m_footWidth / 2, +m_footLength / 2, 0;
+    m_p3 << -m_footWidth / 2, -m_footLength / 2, 0;
+    m_p4 << +m_footWidth / 2, -m_footLength / 2, 0;
+
+    Eigen::Vector3d desiredTranslation = m_input.desiredFootPose.translation();
+    Eigen::Matrix3d desiredRotation = m_input.desiredFootPose.rotation();
+    Eigen::Vector3d desiredRPY = desiredRotation.eulerAngles(2, 1, 0);
+
+    //double desiredYaw = atan2(desiredRotation(2,1), desiredRotation(2,2));
+
+    Eigen::Matrix3d measuredRotation = m_input.measuredRotation.rotation();
+    Eigen::Vector3d measuredRPY = measuredRotation.eulerAngles(2, 1, 0);
+
+    //log()->error("desiredRPY {} fine", desiredRPY);
+    //log()->error("measuredRPY{} fine", measuredRPY);
+
+    measuredRPY(0) = desiredRPY(0);
+
+    //m_measuredRotationCorrected << cos(m_measuredRPY(0)) * cos(m_measuredRPY(1)), cos(m_measuredRPY(0)) * sin(m_measuredRPY(1)) * sin(m_measuredRPY(2)) - sin(m_measuredRPY(0)) * cos(m_measuredRPY(2)), cos(m_measuredRPY(0)) * sin(m_measuredRPY(1)) * cos(m_measuredRPY(2)) + sin(m_measuredRPY(0)) * sin(m_measuredRPY(2)),
+    //                               sin(m_measuredRPY(0)) * cos(m_measuredRPY(1)), sin(m_measuredRPY(0)) * sin(m_measuredRPY(1)) * sin(m_measuredRPY(2)) + cos(m_measuredRPY(0)) * cos(m_measuredRPY(2)), sin(m_measuredRPY(0)) * sin(m_measuredRPY(1)) * cos(m_measuredRPY(2)) - cos(m_measuredRPY(0)) * sin(m_measuredRPY(2)),
+    //                               -sin(m_measuredRPY(1)), cos(m_measuredRPY(1)) * sin(m_measuredRPY(2)), cos(m_measuredRPY(1)) * cos(m_measuredRPY(2));
+
+    m_measuredRotationCorrected = manif::SO3d(measuredRPY(2), measuredRPY(1), measuredRPY(0));  // Check the order is correct
+
+    //log()->error("The matrix {} before", m_input.desiredFootPose.quat());
+    //log()->error("The matrix {} after", m_measuredRotationCorrected);
+
+    manif::SE3d T_foot_raw(desiredTranslation, m_measuredRotationCorrected);
+
+    Eigen::Vector3d p1_intermediate = T_foot_raw.act(m_p1);
+    Eigen::Vector3d p2_intermediate = T_foot_raw.act(m_p2);
+    Eigen::Vector3d p3_intermediate = T_foot_raw.act(m_p3);
+    Eigen::Vector3d p4_intermediate = T_foot_raw.act(m_p4);
+
+    Eigen::Vector4d vertexQuotes(p1_intermediate(2), p2_intermediate(2), p3_intermediate(2), p4_intermediate(2));
+
+    int minQuoteIndex = std::distance(std::begin(vertexQuotes), std::min_element(std::begin(vertexQuotes), std::end(vertexQuotes)));
```

Comment:
Now that I'm reviewing this PR I noticed that in theory this code can run also if the number of corners is different from 4 indeed assumes that you stored all the corners in a `std::vector<Eigen::Vector3d> cornersInLocalFrame`

then you can rotate them with
```cpp
std::vector<Eigen::Vector3d> footCorners;

// for each corner we compute the position in the inertial frame
for (const auto& corner : corners)
{
     footCorners.emplace_back(T_foot_raw.act(corner.position));
}
```
then we can compute the lower corner as 
```cpp
auto lowerCorner = std::min_element(footCorners.begin(),
                                    footCorners.end(),
                                    [](const Eigen::Vector3d& a, const Eigen::Vector3d& b) {
                                        return a[2] < b[2];
                                    });
```
finally, you get the index of the corner as     
```cpp
int supportCornerIndex = std::distance(footCorners.begin(), lowerCorner);
```